### DISCUSSION TITLE
Switching tracks in a connected WebRTC stream

### DISCUSSION THREAD

**Discussion by Manfred** [OvenMediaLabs/OvenMediaEngine#1712] [Help] [answered]
I'm trying to build a streaming booth for educational purposes and I would like users to be able to switch between video inputs and also from video to display media and back without dropping the stream for viewers.

In the [Demo application](https://demo.ovenplayer.com/demo_input.html) it looks like the stream disconnects when I change any of the stream parameters. I can't determine from the documentation is this is a limitation of the JavaScript implementation or OvenMediaEngine.

Would it be technically possible to use replaceTrack on the sender in the RPCPeerConnection without dropping the connection? Is there a different approach that would work with OvenMediaEngine?

I'm not fishing for a full implementation by the way, I'm just trying to figure out if this is a dead end.

**Comment by getroot** [comment, marked as answer]
You can achieve your requirement with scheduled channels.
https://airensoft.gitbook.io/ovenmediaengine/live-source/scheduled-channel

This allows you to switch media sources with an API.
